services/src/handlers/workflows.rs

```diff
@@ -523,9 +477,167 @@ async fn create_dataset<C: Context>(
+                .ok_or(crate::error::Error::MissingSpatialReference)?;
+        let tile_limit = None; // TODO: set a reasonable limit or make configurable?
+
+        // build the geotiff
+        let res = call_on_generic_raster_processor_gdal_types!(processor, p => raster_stream_to_geotiff(
+            &self.file_path,
+            p,
+            query_rect,
+            query_ctx,
+            GdalGeoTiffDatasetMetadata {
+                no_data_value: Default::default(), // TODO: decide how to handle the no data here
+                spatial_reference: request_spatial_ref,
+            },
+            GdalGeoTiffOptions {
+                compression_num_threads: get_config_element::<crate::util::config::Gdal>()?.compression_num_threads,
+                as_cog: self.info.as_cog,
+                force_big_tiff: false,
+            },
+            tile_limit,
+            Box::pin(futures::future::pending()), // datasets shall continue to be built in the background and not cancelled
+        ).await)?
+            .map_err(crate::error::Error::from)?;
+
+        // create the dataset
+        let dataset = create_dataset(
+            self.info.clone(),
+            res,
+            result_descriptor,
+            self.ctx.get_ref(),
+            self.session.clone(),
+        )
+        .await?;
+
+        Ok(RasterDatasetFromWorkflowResult {
+            dataset,
+            upload: self.upload,
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl<C: Context> Task<C::TaskContext> for RasterDatasetFromWorkflowTask<C> {
+    async fn run(
+        &self,
+        _ctx: C::TaskContext,
+    ) -> Result<Box<dyn crate::tasks::TaskStatusInfo>, Box<dyn ErrorSource>> {
+        let response = self.process().await;
+
+        response
+            .map(TaskStatusInfo::boxed)
+            .map_err(ErrorSource::boxed)
+    }
+
+    async fn cleanup_on_error(&self, _ctx: C::TaskContext) -> Result<(), Box<dyn ErrorSource>> {
+        fs::remove_dir_all(&self.file_path)
+            .await
+            .context(crate::error::Io)
+            .map_err(ErrorSource::boxed)?;
+
+        //TODO: Dataset might already be in the database, if task was already close to finishing.
+
+        Ok(())
+    }
+
+    fn task_type(&self) -> &'static str {
+        "create-dataset"
+    }
+
+    fn task_unique_id(&self) -> Option<String> {
+        Some(self.upload.to_string())
+    }
+}
+
+/// Create a task for creating a new dataset from the result of the workflow given by its `id` and the dataset parameters in the request body.
+/// Returns the id of the created task
+#[utoipa::path(
+tag = "Workflows",
+post,
+path = "/datasetFromWorkflowTask/{id}",
+request_body = RasterDatasetFromWorkflow,
+responses(
+(status = 200, description = "Id of created task", body = TaskResponse,
+example = json!({"task_id": "7f8a4cfe-76ab-4972-b347-b197e5ef0f3c"})
+)
+),
+params(
+("id" = WorkflowId, description = "Workflow id")
+),
+security(
+("session_token" = [])
+)
+)]
```

Comment:
```suggestion
#[utoipa::path(
    tag = "Workflows",
    post,
    path = "/datasetFromWorkflowTask/{id}",
    request_body = RasterDatasetFromWorkflow,
    responses(
        (status = 200, description = "Id of created task", body = TaskResponse,
        example = json!({"task_id": "7f8a4cfe-76ab-4972-b347-b197e5ef0f3c"})
        )
    ),
    params(
        ("id" = WorkflowId, description = "Workflow id")
    ),
    security(
        ("session_token" = [])
    )
)]
```

There seems to be no auto formatted, but maybe we can do it accordingly.